src/Api/DataTransformer/GiftCardCodeAwareInputCommandDataTransformer.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusGiftCardPlugin\Api\DataTransformer;
+
+use Setono\SyliusGiftCardPlugin\Api\Command\GiftCardCodeAwareInterface;
+use Setono\SyliusGiftCardPlugin\Model\GiftCardInterface;
+use Sylius\Bundle\ApiBundle\DataTransformer\CommandDataTransformerInterface;
+
+final class GiftCardCodeAwareInputCommandDataTransformer implements CommandDataTransformerInterface
+{
+    /**
+     * @param GiftCardCodeAwareInterface $object
+     */
+    public function transform($object, string $to, array $context = []): GiftCardCodeAwareInterface
+    {
+        /** @var GiftCardInterface $giftCard */
+        $giftCard = $context['object_to_populate'];
+
+        $object->setGiftCardCode($giftCard->getCode());
+
+        return $object;
+    }
+
+    /**
+     * @param object $object
+     */
```

Comment:
Do we need this phpdoc?

Reply:
Yeah, otherwise Psalm throws MissingParamType

Reply:
Oh yeah, it doesn't like the generic types... Just as with `mixed`...